electronics/charging_capacitor.py

```diff
@@ -0,0 +1,114 @@
+    resistance: float,  # resistance in ohms.
+    capacitance: float,  # capacitance in farads.
+    time_sec: float,  # time in seconds after charging initiation of capacitor.
+):
+    """
+    find voltage of capacitor at any nth second after the initiation of it's charging.
+
+    Parameters
+    ----------
+    source_voltage : float
+        it is going to multiply with rest of the function.
+    resistance : float
+        it is using in RC function.
+    capacitance : float
+        it is multiplying with resistance_ohms to yield output.
+    time_sec : float
+        it is dividing by RC.To find the voltage at nth second.
+
+    Examples
+    --------
+    >>> charging_capacitor(source_voltage=15,resistance=200,capacitance=20,time_sec=2)
+    0.007
+
+    >>> charging_capacitor(source_voltage=0,resistance=1000,capacitance=30,time_sec=3)
+    Traceback (most recent call last):
+        ...
+    ValueError: source voltage cannot be zero.
+
+    >>> charging_capacitor(20,2000,30*pow(10,-5),4)
+    19.975
+
+    >>> charging_capacitor(source_voltage=20,resistance=-2000,capacitance=30,time_sec=4)
+    Traceback (most recent call last):
+        ...
+    ValueError: Resistance cannot be negative.
+
+    >>> charging_capacitor(source_voltage=-2,resistance=20,capacitance=30,time_sec=4)
+    Traceback (most recent call last):
+        ...
+    ValueError: source voltage cannot be negative.
+
+    >>> charging_capacitor(source_voltage=8,resistance=0,capacitance=30,time_sec=4)
+    Traceback (most recent call last):
+        ...
+    ValueError: Resistance cannot be zero.
+
+    >>> charging_capacitor(source_voltage=30,resistance=1500,capacitance=0,time_sec=4)
+    Traceback (most recent call last):
+        ...
+    ValueError: Capacitance cannot be zero.
+
+    >>> charging_capacitor(source_voltage=30,resistance=23,capacitance=-40,time_sec=5)
+    Traceback (most recent call last):
+        ...
+    ValueError: Capacitance cannot be negative.
+    """
+
+    if source_voltage <= 0:
+        if source_voltage < 0:
+            raise ValueError("source voltage cannot be negative.")
+        elif source_voltage == 0:
+            raise ValueError("source voltage cannot be zero.")
+        else:
+            return None
+    elif resistance <= 0:
+        if resistance < 0:
+            raise ValueError("Resistance cannot be negative.")
+        elif resistance == 0:
+            raise ValueError("Resistance cannot be zero.")
+        else:
+            return None
+    elif capacitance <= 0:
+        if capacitance < 0:
+            raise ValueError("Capacitance cannot be negative.")
+        elif capacitance == 0:
+            raise ValueError("Capacitance cannot be zero.")
+        else:
+            return None
+    else:
+        if resistance != 0:
+            return round(
+                source_voltage * (1 - exp(-time_sec / (resistance * capacitance))), 3
+            )
+        else:
+            return None
```

Comment:
`None` is not a voltage but `0` is a voltage.

Reply:
DONE!